src/CommunityToolkit.Mvvm.CodeFixers/ClassUsingAttributeInsteadOfInheritanceCodeFixer.cs

```diff
@@ -101,35 +97,13 @@ private static async Task<Document> UpdateReference(Document document, ClassDecl
                     (identifierName == attributeTypeName || (identifierName + "Attribute") == attributeTypeName))
                 {
                     // We found the attribute to remove and the list to update
-                    targetAttributeList = attributeList;
-                    targetAttribute = attribute;
+                    updatedClassDeclaration = (ClassDeclarationSyntax)generator.RemoveNode(updatedClassDeclaration, attribute);
```

Comment:
I assume this will also take care of removing the whole attribute list if it becomes empty?

Reply:
@Sergio0694 Yes. Roslyn implementation of this part:

https://github.com/dotnet/roslyn/blob/13c1bfbdece3f3890655c5944928e3952189ee05/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs#L2994-L3002